frame/examples/offchain-worker-ping-pong/src/lib.rs

```diff
@@ -0,0 +1,681 @@
+		let next_unsigned_at = <NextUnsignedAt<T>>::get();
+		if &next_unsigned_at > block_number {
+			return InvalidTransaction::Stale.into()
+		}
+		// Let's make sure to reject transactions from the future.
+		let current_block = <system::Pallet<T>>::block_number();
+		if &current_block < block_number {
+			return InvalidTransaction::Future.into()
+		}
+
+		ValidTransaction::with_tag_prefix("ExampleOffchainWorker")
+			// We set base priority to 2**20 and hope it's included before any
+			// other transactions in the pool.
+			.priority(T::UnsignedPriority::get().saturating_add(0))
+			// This transaction does not require anything else to go before into
+			// the pool. In theory we could require `previous_unsigned_at`
+			// transaction to go first, but it's not necessary in our case.
+			//.and_requires() We set the `provides` tag to be the same as
+			// `next_unsigned_at`. This makes sure only one transaction produced
+			// after `next_unsigned_at` will ever get to the transaction pool
+			// and will end up in the block. We can still have multiple
+			// transactions compete for the same "spot", and the one with higher
+			// priority will replace other one in the pool.
+			.and_provides(next_unsigned_at)
+			// The transaction is only valid for next 5 blocks. After that it's
+			// going to be revalidated by the pool.
+			.longevity(5)
+			// It's fine to propagate that transaction to other peers, which
+			// means it can be created even by nodes that don't produce blocks.
+			// Note that sometimes it's better to keep it for yourself (if you
+			// are the block producer), since for instance in some schemes
+			// others may copy your solution and claim a reward.
+			.propagate(true)
+			.build()
+	}
+
+	/// A helper function to send a signed pong transaction from the OCW.
+	fn ocw_pong_signed() -> Result<(), &'static str> {
+		let signer = Signer::<T, T::AuthorityId>::all_accounts();
+		if !signer.can_sign() {
+			return Err(
+				"No local accounts available. Consider adding one via `author_insertKey` RPC.",
+			)
+		}
+
+		let pings = <Pings<T>>::get();
+		for p in pings {
+			let Ping(nonce) = p;
+
+			// Using `send_signed_transaction` associated type we create and
+			// submit a transaction representing the call, we've just created.
+			// Submit signed will return a vector of results for all accounts
+			// that were found in the local keystore with expected `KEY_TYPE`.
+			let results = signer.send_signed_transaction(|_account| {
+				// nonce is wrapped into a call to `pong_signed` public function
+				// of this pallet. This means that the transaction, when
+				// executed, will simply call that function passing `nonce` as
+				// an argument.
+				Call::pong_signed { nonce }
+			});
+
+			for (acc, res) in &results {
+				match res {
+					Ok(()) => log::info!("[{:?}] Submitted pong with nonce {}", acc.id, nonce),
+					Err(e) => log::error!("[{:?}] Failed to submit transaction: {:?}", acc.id, e),
```

Comment:
Why is this not propagating the error?

Reply:
63b1029df0599bef5ba463ead38b690470317fd2